app/javascript/packages/webauthn/extract-credentials.ts

```diff
@@ -0,0 +1,15 @@
+import { base64ToArrayBuffer } from './converters';
+
+/**
+ * Converts an array of base64-encoded strings to credential descriptors.
+ *
+ * @param credentials Strings to convert.
+ * @return Converted credentials.
+ */
+const extractCredentials = (credentials: string[]): PublicKeyCredentialDescriptor[] =>
```

Comment:
The authentication flow also uses this function. For LG-9873 / #8639, we'll need to find a way to include `transports` as part of the mapping. I don't think it needs to be addressed here, but surfacing for visibility.

Reply:
I'm thinking maybe we'd eventually get rid of `extractCredentials`, and push that to the individual call-sites. #8639 will need to restructure how `credential_ids` are passed to authentication anyways, since it's no longer just a simple comma-separated array of base64 strings, we'd need some way to do structured data of `[{ credentialId, transports }]`.